llvm/lib/Target/AArch64/AArch64Processors.td

```diff
@@ -1010,6 +1034,9 @@ def : ProcessorModel<"apple-a16", CycloneModel, ProcessorFeatures.AppleA16,
                      [TuneAppleA16]>;
 def : ProcessorModel<"apple-a17", CycloneModel, ProcessorFeatures.AppleA17,
                      [TuneAppleA17]>;
+def : ProcessorModel<"apple-m4", CycloneModel, ProcessorFeatures.AppleM4,
```

Comment:
Shouldn't this go below, under `// Mac CPUs`?

Reply:
No, it is an iPad CPU:

https://www.apple.com/newsroom/2024/05/apple-introduces-m4-chip/

Reply:
We should probably rearrange them to remove that distinction though, so that aliases that share the same `ProcessorFeatures.AppleA*` are together. I'll do that in a follow-up commit.

Reply:
Ah, right - sorry I forgot about that distinction. (I guess it'll end up in macs at some point too, and I know you can't comment on that - so until then this is indeed the right sorting.)

Reply:
https://github.com/llvm/llvm-project/pull/95579